skipruntime-ts/core/native/src/Extern.sk

```diff
@@ -295,6 +313,32 @@ class ExternResourceBuilder(
   }
 }
 
+/************  Checker ****************/
```

Comment:
I'm a bit confused about what these are doing and why -- would you explain briefly?

I see the OneChecker and AllChecker over in skipruntime_module.ts which make queries and check reject/resolve depending on the number of errors.  But this seems very heavy weight to set up all these interfaces and pass things across the FFI to do so, right?

And, with all this indirection I'm not positive, but it seems to me that the getOne call within OneChecker will itself create a new OneChecker -- where/how does that get deduplicated?

Reply:
No, the “OneChecker” doesn't itself create a “OneChecker”, the “Checker” calls the corresponding “get” with a request identifier in order to retrieve the corresponding state.
Perhaps this could be simplified. I did it this way because I didn't want to be too closely modeled on the way promises work. I'd rather wait until I've integrated other languages before optimizing this.
I started by managing resolve/reject directly, but in terms of type and FFI management, it was heavier than this version.